### ISSUE TITLE
geos_c.h gives -Wstrict-prototypes warnings

### ISSUE THREAD

**Issue by dmacks** [libgeos/geos#437]
Using geos-3.8.2 built on OS X 10.13, compiling the following foo.c:
```
#include <geos_c.h>
void foo(void) {
  GEOSBufferParams* bp;
  bp = GEOSBufferParams_create();
  GEOSBufferParams_destroy(bp);
}
```
using:
```
gcc -c foo.c -I/sw/opt/libgeos3.8.2/include -Wall -Wstrict-prototypes
```
gives a ton of warnings like:
```
In file included from foo.c:1:
/sw/opt/libgeos3.8.2/include/geos_c.h:160:37: warning: this function declaration
      is not a prototype [-Wstrict-prototypes]
typedef void (GEOSInterruptCallback)();
                                    ^
                                     void
/sw/opt/libgeos3.8.2/include/geos_c.h:163:43: warning: this function declaration
      is not a prototype [-Wstrict-prototypes]
extern void GEOS_DLL GEOS_interruptRequest();
                                          ^
                                           void
```
If the function takes no parameters, clang wants it explicitly prototyped thay way, for example:
```
extern void GEOS_DLL GEOS_interruptRequest(void);
```

**Comment by macdrevx** [comment]
Would love to see this fixed. We apply a little [patch](https://github.com/GEOSwift/geos/blob/main/update.sh#L31-L35) to work around this when packaging geos for GEOSwift.

**Comment by pramsey** [comment]
I've fixed this back to 3.8.
